packages/rest/src/lib/RequestManager.ts

```diff
@@ -406,12 +415,15 @@ export class RequestManager extends EventEmitter {
 
 		const fetchOptions = {
 			agent: this.agent,
-			body: finalBody,
 			// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
 			headers: { ...(request.headers ?? {}), ...additionalHeaders, ...headers } as Record<string, string>,
 			method: request.method,
 		};
 
+		if (finalBody !== undefined) {
+			Reflect.set(fetchOptions, 'body', finalBody);
```

Comment:
Assuming the reason you're using `Reflect.set()` here is because of the type, can't you just explicitly type `fetchOptions` as `RequestInit`?

Reply:
This is my lazy temporary solution because undici's fetch doesn't support an `agent` option.

Reply:
fixed in d9dd20e55606643ffab1bf88a7802815458d9bd6